pkg/comp-functions/functions/vshnpostgres/maintenance.go

```diff
@@ -202,3 +204,20 @@ func setPsqlMinorVersion(svc *runtime.ServiceRuntime, desiredCluster *stackgresv
 
 	return nil
 }
+
+func addStackgresCredentialsObserver(svc *runtime.ServiceRuntime, comp *vshnv1.VSHNPostgreSQL, sgNamespace string) error {
```

Comment:
In my PR of major upgrade I created a kube client to get this secret :(

Reply:
Please don't create a kube client, use observers. Kube clients will open a whole new world of PITA for non-converged setups...

Reply:
here it is my comment :D

Reply:
Remember this thing, you will have to rebase on my PR or on master if I am able to deploy it soon